manifests/entry.pp

```diff
@@ -3,7 +3,7 @@
   $options,
   $ensure = $backupninja::ensure,
   $weight = '',
-  $when   = '',
+  Array[String] $when   = [],
```

Comment:
Please let's try not breaking current API, something like `Variant[Array[String], String` should work.

Also if we are Typing params we should type all in a later commit/PR (for clarity)

Reply:
Understood